repl.html

```diff
@@ -120,38 +120,45 @@
     input.addEventListener("keyup", e => inputHandler(e));
 }
 
+async function submitInput() {
+    const response = await sendRequest(document.env, input.value);
+    if (response.ok) {
+        const {env, result} = await response.json();
+        updateHistory(input.value, result);
+        history.push([input.value, result]);
+        input.value = "";
+        document.env = env;
+        window.localStorage.setItem('env', env)
+        window.localStorage.setItem('history', JSON.stringify(history));
+    } else {
+        const msg = await response.text();
+        updateHistory(input.value, msg);
+        input.value = "";
+    }
+}
+
 loadFromLocalStorage();
 input.focus();
 async function inputHandler(e) {
     // TODO(max): Make up/down arrow keys navigate history
     if (e.key === "Enter") {
-        if (e.shiftKey && input.tagName === "INPUT") {
+        if (input.tagName === "INPUT" && e.shiftKey) {
             // Shift-Enter expands the input to a textarea and does not submit
             // input.
             expandInput();
             return;
         }
-        if (!e.shiftKey && input.tagName === "TEXTAREA") {
-            // Normal Enter in a textarea should not submit input.
-            return;
-        }
-        const response = await sendRequest(document.env, input.value);
```

Comment:
Any particular reason to extract into a function?

Reply:
We want to use it in both on enter and on submit press

Reply:
oic